### PR TITLE
mmwtex.c: boolean-style evaluation in conditionals (+ linting)

### PR DESCRIPTION
**Pull request by benjub**
Found some inconsistencies while on #160, which I signal as TODOs in this PR. I also made some simplifications (mainly, evaluate boolean-like variables as bools) and linting, all in `mmwtex.c`.

### CODE REVIEW

File: src/mmwtex.c
Comment (tirix) [inline src/mmwtex.c:33]: Unused? What about https://github.com/metamath/metamath-exe/blob/ae35c1b2931be8f9683f8bf66d1d10c9b56db706/src/metamath.c#L2854
And https://github.com/metamath/metamath-exe/blob/ae35c1b2931be8f9683f8bf66d1d10c9b56db706/src/metamath.c#L2616

Comment (digama0) [reply src/mmwtex.c:33]: And
https://github.com/metamath/metamath-exe/blob/ae35c1b2931be8f9683f8bf66d1d10c9b56db706/src/metamath.c#L2578-L2587
which shows that it is not a constant either.

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:62]: nitpick: Does metamath-exe actually produce LaTeX (as compared to TeX) output? I would not be surprised if it is plain TeX.

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:856]: This is incorrect, `bug` can return! The user is provided a prompt and one of the options is to continue.

File: src/mmwtex.c
Comment (benjub) [reply src/mmwtex.c:33]: Should I move the declaration/initialization `flag g_briefHtmlFlag = 0;` to `metamath.c` ?

File: src/mmwtex.c
Comment (benjub) [reply src/mmwtex.c:856]: So, I restore `return 0;` but remove its comment ?

File: src/mmwtex.c
Comment (benjub) [reply src/mmwtex.c:62]: `metamath-exe` produces LaTeX code that is not TeX-compatible, since it outputs "\documentclass" for instance, and uses several packages.

File: src/mmwtex.c
Comment (digama0) [reply src/mmwtex.c:33]: I think it could even be moved to a local variable in `command()`, one less global variable to worry about.

Comment (benjub) [reply src/mmwtex.c]: done in 6bc25466c16958979ad8ab106381c3049d086606

File: src/mmwtex.c
Comment (benjub) [reply src/mmwtex.c]: done in 811908c3ef9857018473e9d66022cda45bc0e6dd

File: src/mmwtex.c
Comment (benjub) [inline src/mmwtex.c]: Am I correct ? This follows a `while (1)`  loop with no `break;` statement in the outermost loop (the `break;` is in a `switch` statement within the `while(1)` loop). Therefore, this is unreachable code.

Comment (digama0) [reply src/mmwtex.c]: The `break;` after `return modeSection;` looks very strange in that code, that is also clearly unreachable code.

Comment (benjub) [reply src/mmwtex.c]: Done in 6523fc7c1a292fdbba145b11d65a5d2f876e518f
I didn't go further and didn't change the `switch` statement since I don't know the code enough (`mode` seems to take "magic" values, for instance).

**Review by tirix — APPROVED** [review]

**Comment by benjub** [comment]
@digama0 : following approval by Thierry, can you merge this one ?

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:36]:
```suggestion
// Explorer extension to the set.mm database.
```

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:70]:
```suggestion
// LaTeX output file
```

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:199]:
```suggestion
    if (!g_htmlFlag /* LaTeX */ && g_altHtmlFlag) {
```

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:1097]:
```suggestion
// Converts a comment section in math mode to LaTeX.  Each math token
```
I think indicative mood is more conventional for documentation and is already used elsewhere to my knowledge

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:1652]:
```suggestion
// Prints an embedded comment in LaTeX or HTML.  The commentPtr must point to the first
```

File: src/mmwtex.c
Comment (digama0) [inline src/mmwtex.c:4656]:
```suggestion
// Returns HTML for the pink number to print after the statement labels
```

**Review by digama0 — APPROVED** [review]

**Comment by benjub** [comment]
I didn't apply the four suggestions changing imperative to indicative since I thought it was the opposite convention, but I'm unsure. I'll let you decide to apply or not before merging.

**Comment by digama0** [comment]
I took a census of how this is used in the .h file comments on all functions. As expected, there has not historically been any consistency about this, but the indicatives seem to be winning:

* Imperative mood: 73 functions
* Indicative mood: 94 functions
* Noun phrase: 31 functions

Doxygen seems to have thrown its opinion into the ring slightly with the `\returns` keyword. My preference is to standardize on indicative mood, perhaps it should be added as an additional element to the "code cleanup" issue?

**Comment by benjub** [comment]
> perhaps it should be added as an additional element to the "code cleanup" issue?

Done. (Add suggestions applied.)

**Pull request merged**
